Engine/Nevermore.module.lua

```diff
@@ -36,79 +36,52 @@ local function GetFirstChild(Parent, Name, Class) -- This is what allows the cli
 
 	return Object, Bool
 end
-local Retrieve = GetFirstChild
+local Retrieve, LocalResourcesLocation = GetFirstChild
 
 local function Error(Parent, Name, Class)
 	return Parent:FindFirstChild(Name) or error(("[Nevermore] %s \"%s\" is not installed."):format(Class, Name))
 end
 
-local Nevermore = {
-	__metatable = "[Nevermore] Nevermore's metatable is locked";
-	GetFirstChild = GetFirstChild;
-}
-
-local function GetFolder() -- Placeholder for first time `CreateResourceManager` runs; gets overwritten
-	return GetFirstChild(ResourcesLocation, "Resources", "Folder")
-end
-
-local LocalResourcesLocation
-local function GetLocalFolder() -- Doesn't load by default on the Client
-	return Retrieve(LocalResourcesLocation, "Resources", "Folder")
-end
-
+local Nevermore = {GetFirstChild = GetFirstChild}
 local SmartFolder = {}
+
 function SmartFolder:__call(this, Name, Parent)
 	if this ~= Nevermore then -- Enables functions to support calling by '.' or ':'
 		Name, Parent = this, Name
 	end
-	local Table = self.Table
-	local Object, Bool = Table[Name], false
+	local Contents = self.Contents
+	local Object, Bool = Contents[Name]
 	if not Object then
-		local Folder = self.Folder
+		local Folder, Class = self.Folder, self.Class
 		if not Folder then
-			Folder = self.GetFolder(self.FolderName)
+			Folder = self.GetFolder(Plurals[Class] or Class .. "s")
 			self.Folder = Folder
 		end
-		Object, Bool = self.Retrieve(Parent or Folder, Name, self.Class)
+		Object, Bool = self.Retrieve(Parent or Folder, Name, Class)
 		if not Parent then
-			Table[Name] = Object
+			Contents[Name] = Object
 		end
 	end
-	return Object, Bool
+	return Object, Bool or false
 end
 
-local function CreateResourceManager(Nevermore, FullName, Data) -- Create methods called to Nevermore
-	if type(FullName) == "string" then
-		local GetFirstChild = GetFirstChild
-		local Name, Local = FullName:gsub("^Get", ""):gsub("^Local", "")
-		local Class = Classes[Name] or Name
-		local ResourceManager = {
-			Class = Class;
-			Table = Data or {};
-			FolderName = Plurals[Class] or Class .. "s";
-			Retrieve = GetFirstChild;
-		}
-
-		if Local == 0 then
-			ResourceManager.GetFolder = GetFolder
-		else
-			ResourceManager.GetFolder = GetLocalFolder
-			GetFirstChild = Retrieve
-		end
-
-		if GetFirstChild == Retrieve then
-			local Success, Object = pcall(Instance.new, Class)
-			ResourceManager.Retrieve = Success and not Object:Destroy() and GetFirstChild or Error
-		end
-
-		Nevermore[FullName] = ResourceManager
-		return setmetatable(ResourceManager, SmartFolder)
-	else
-		error("[Nevermore] Attempt to index Nevermore with invalid key: string expected, got " .. type(FullName))
-	end
-end
-GetFolder = CreateResourceManager(Nevermore, "GetFolder")
-GetLocalFolder = CreateResourceManager(Nevermore, "GetLocalFolder")
```

Comment:
GetLocalFolder is for instancing local storage on either the client or server. On the client, this is the LocalPlayer.Resources folder. On the server, this is the ServerStorage.Resources folder